daemon/libnetwork/internal/resolvconf/resolvconf.go

```diff
@@ -291,70 +289,108 @@ func (rc *ResolvConf) TransformForIntNS(
 // is true, the file will include header information if supplied, and a trailing
 // comment that describes how the file was constructed and lists external resolvers.
 func (rc *ResolvConf) Generate(comments bool) ([]byte, error) {
-	s := struct {
-		Md          *metadata
-		NameServers []netip.Addr
-		Search      []string
-		Options     []string
-		Other       []string
-		Overrides   []string
-		Comments    bool
-	}{
-		Md:          &rc.md,
-		NameServers: rc.nameServers,
-		Search:      rc.search,
-		Options:     rc.options,
-		Other:       rc.other,
-		Comments:    comments,
+	var b bytes.Buffer
+	b.Grow(512) // estimated size for a regular resolv.conf we produce.
+
+	if comments && rc.md.Header != "" {
+		b.WriteString(rc.md.Header + "\n")
+		b.WriteByte('\n')
```

Comment:
Probably not worth changing, but ...

```suggestion
		b.WriteString(rc.md.Header)
		b.WriteString("\n\n")
```